### PR TITLE
Compatibility with pushgateway 0.4

### PR DESCRIPTION
**Pull request by RubenSlabbert**
As of pushgateway 0.4, timestamps are no longer accepted when pushing.

See: https://github.com/prometheus/pushgateway/pull/119

### CODE REVIEW

**Comment by siimon** [comment]
Maybe a silly question, but why do you have timestamps in your metrics in the first place when you know it's not supported by the pushgateway?

**Comment by RubenSlabbert** [comment]
As far as I could gather, the default metrics collection automatically adds timestamps to guages and counters, which will then automatically be pushed to the gateway.

**Comment by siimon** [comment]
Gotcha!

Add some test cases and update the README + Changelog :)

File: lib/registry.js [hunk 1]
```diff
@@ -48,9 +57,9 @@ class Registry {
 		return acc;
 	}
 
-	metrics() {
+	metrics(opts) {
 		return this.getMetricsAsArray()
-			.map(this.getMetricAsPrometheusString, this)
+			.map(metric => this.getMetricAsPrometheusString.bind(this)(metric, opts))
```
Comment (SimenB) [inline lib/registry.js]: Don't use bind, just call it normally. The arrow function ensures correct `this`

**Comment by RubenSlabbert** [comment]
All the requested changes have been implemented. Let me know if you'd like more tests.

**Comment by RubenSlabbert** [comment]
Works perfectly for me with [pushgateway 0.4](https://github.com/prometheus/pushgateway/releases/tag/v0.4.0).

**Comment by siimon** [comment]
Great! Thanks! :)

**Comment by SimenB** [comment]
Published as 10.0.3.

Thanks!

**Pull request merged**